demos/rag_agentic/notebooks/Level6_agents_MCP_and_RAG.ipynb

```diff
@@ -560,20 +562,304 @@
     "6. Finally, the LLM sends the summarization as a message on Slack using the `slack_post_message` tool call configured with the Slack MCP server."
    ]
   },
+  {
+   "cell_type": "markdown",
+   "id": "39f6c84d",
+   "metadata": {},
+   "source": [
+    "## Defining our Agent - ReAct\n",
+    "\n",
+    "Now that we've shown that we can successfully accomplish this multi-step multi-tool task using prompt chaining, let's see if we can give our agent a bit more autonomy to perform the same task but with a single prompt instead of a chian. To do this, we will instantiate a ReAct agent (which is included in the llama stack python client by default). The ReAct agent is a variant of the simple agent but with the ability to loop through \"Reason then Act\" iterations, thinking through the problem and then using tools until it determines that it's task has been completed successfully.\n",
```

Comment:
probably worth to follow/reuse the ReAct definition mentioned in the Level 3 notebook